sympy/matrices/matrices.py

```diff
@@ -877,6 +877,136 @@ def rank(self, iszerofunc=_iszero, simplify=False):
         echelon_form, pivots, swaps = mat._eval_echelon_form(iszerofunc=iszerofunc, simpfunc=simpfunc)
         return len(pivots)
 
+    def rank_decomposition(self,
+        iszerofunc=_iszero, simplify=False, format='original'):
+        r"""Returns the rank-factorized form of `A`, as `C` and `F`.
```

Comment:
How about `Returns a pair of matrices (C,F) so that A=C*F and both C and F have the same rank as A.